test/topology_custom/test_commitlog_segment_data_resurrection.py

```diff
@@ -0,0 +1,124 @@
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+import os
+import logging
+import glob
+import json
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_pinned_cl_segment_doesnt_resurrect_data(manager: ManagerClient):
+    """
+        The tested scenario is as follows:
+        * Two tables, ks1.tbl1 and ks2.tbl2.
+        * A commitlog segment contains writes for both tables.
+        * ks1.tbl1 has very low write traffix, memtables are very rarely
+          flushed, this results in ks1.tbl1 "pinning" any commitlog segment
+          which has writes, that are still in the memtable.
+        * ks2.tbl2 deletes the data, some of which is still in the segment
+          pinned by ks1.tbl1.
+        * ks2.tbl2 flushes the memtable, data gets to sstables, commitlog
+          segments containing writes for both tables are still pinned.
+        * ks2.tbl2 compacts and purges away both the data and the tombstone.
+        * The node has an unclean restart, the pinned segments are replayed and
+          data is resurrected, as the tombstone is already purged.
+
+        This test uses gc_grace_seconds=0 to make the test fast.
+
+        See https://github.com/scylladb/scylladb/issues/14870
+        """
+    server = await manager.server_add(config={
+        "commitlog_sync": "batch",
+        "commitlog_segment_size_in_mb": 1,
```

Comment:
Are these commitlog configurations necessary for this test or somehow allow to make it smaller? If not, it's better not to have them at all.

Reply:
Reading the code below, I am beginning to understand that your scenario needs (even though it's not mentioned in the comment above) to fill a whole segment, so it's good for the test speed to have a small segment size, lower than the default. So I agree it's good to lower it. I still don't know what "commitlog_sync: batch" is about.

Reply:
I as wrote in CR description I just refactored existing test to be executed in community version and tried to keep it as close to the original one as possible. Concerning commitlog segment size - it's very important thing. To make test faster.
But I tried to use default commitlog_sync mode and found to difference. However it would be nice to ask author of original test to be sure